dotnet/src/IntegrationTests/Connectors/OpenAI/ChatHistoryTests.cs

```diff
@@ -0,0 +1,133 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+using SemanticKernel.IntegrationTests.TestSettings;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SemanticKernel.IntegrationTests.Connectors.OpenAI;
+
+public sealed class ChatHistoryTests : IDisposable
+{
+    private readonly IKernelBuilder _kernelBuilder;
+    private readonly XunitLogger<Kernel> _logger;
+    private readonly RedirectOutput _testOutputHelper;
+    private readonly IConfigurationRoot _configuration;
+    private static readonly JsonSerializerOptions s_jsonOptionsCache = new() { WriteIndented = true };
+    public ChatHistoryTests(ITestOutputHelper output)
+    {
+        this._logger = new XunitLogger<Kernel>(output);
+        this._testOutputHelper = new RedirectOutput(output);
+        Console.SetOut(this._testOutputHelper);
+
+        // Load configuration
+        this._configuration = new ConfigurationBuilder()
+            .AddJsonFile(path: "testsettings.json", optional: false, reloadOnChange: true)
+            .AddJsonFile(path: "testsettings.development.json", optional: true, reloadOnChange: true)
+            .AddEnvironmentVariables()
+            .AddUserSecrets<OpenAICompletionTests>()
+            .Build();
+
+        this._kernelBuilder = Kernel.CreateBuilder();
+    }
+
+    [Fact]
+    public async Task ItSerializesAndDeserializesChatHistoryAsync()
+    {
+        // Arrange
+        this._kernelBuilder.Services.AddSingleton<ILoggerFactory>(this._logger);
+        var builder = this._kernelBuilder;
+        this.ConfigureAzureOpenAIChatAsText(builder);
+        builder.Plugins.AddFromType<FakePlugin>();
+        var kernel = builder.Build();
+
+        OpenAIPromptExecutionSettings settings = new() { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions };
+        ChatHistory history = new();
+
+        history.AddUserMessage("Make me a special poem");
+        var historyBeforeJson = JsonSerializer.Serialize(history.ToList(), s_jsonOptionsCache);
+        var service = kernel.GetRequiredService<IChatCompletionService>();
+        ChatMessageContent result = await service.GetChatMessageContentAsync(history, settings, kernel);
+        history.AddUserMessage("Ok thank you");
+
+        ChatMessageContent resultOriginalWorking = await service.GetChatMessageContentAsync(history, settings, kernel);
+        var historyJson = JsonSerializer.Serialize(history, s_jsonOptionsCache);
+        var historyAfterSerialization = JsonSerializer.Deserialize<ChatHistory>(historyJson);
+        ChatMessageContent shouldNotThrow = await service.GetChatMessageContentAsync(historyAfterSerialization!, settings, kernel);
+    }
+
+    private void ConfigureAzureOpenAIChatAsText(IKernelBuilder kernelBuilder)
+    {
+        var azureOpenAIConfiguration = this._configuration.GetSection("Planners:AzureOpenAI").Get<AzureOpenAIConfiguration>();
```

Comment:
Is this correct configuration for `ChatHistory` tests?

Reply:
Yeah, I need the planners configuration because it uses the gpt-4-preview to work. Older models will break.